libs/payments/iap/src/lib/google/google-iap.service.ts

```diff
@@ -0,0 +1,123 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Injectable, Logger } from '@nestjs/common';
+import {
+  AppendedPlayStoreSubscriptionPurchase,
+  DeveloperNotification,
+  NotificationType,
+} from './types';
+import { GoogleIapPurchaseManager } from './google-iap-purchase.manager';
+import { PlayStoreSubscriptionPurchase } from './subscription-purchase';
+import {
+  GoogleIapConflictError,
+  GoogleIapInvalidPurchaseTokenError,
+} from './google-iap.error';
+
+@Injectable()
+export class GoogleIapService {
+  constructor(
+    private purchaseManager: GoogleIapPurchaseManager,
+    private log: Logger
+  ) {}
+
+  async getActiveSubscriptions(
+    uid: string
+  ): Promise<AppendedPlayStoreSubscriptionPurchase[]> {
+    const allPurchases = await this.purchaseManager.getForUser(uid);
+    const purchases = allPurchases.filter((purchase) =>
+      purchase.isEntitlementActive()
+    );
```

Comment:
This filter already occurs at the manager level

Reply:
Oh, hmm, after double checking it looks like `PurchaseManager.getForUser` includes AccountHold and Paused purchases as well as EntitlementActive.

Lemme see if I can find out why that's the case.

Reply:
Nope, couldn't find a place where purchases on AccountHold or Paused was used anywhere. I'll update the manager to only return purchases with `isEntitlementActive`